File: pcgen-front/app/models/fan.js
import DS from 'ember-data';
import Product from './product';

export default Product.extend({
  bearing: DS.attr('string'),
  minRpm: DS.attr('number'),
  maxRpm: DS.attr('number'),
  startingVoltage: DS.attr('number'),
  maxAmps: DS.attr('number'),
  mtbfHours: DS.attr('number'),
  hasPwm: DS.attr('boolean'),
  highPressure: DS.attr('boolean'),
  quiet: DS.attr('boolean'),
  thicknessMm: DS.attr('number'),
  // coolerFanSlots: DS.hasMany('cooler-fan-slot'),
  // caseFanSlots: DS.hasMany('case-fan-slot'),
  powerSupplies: DS.hasMany('power-supply')
});

File: app/models/fan.rb
class Fan < ApplicationRecord
  validates :max_rpm, :thickness_mm, presence: true

  has_one :property_list, as: :product, dependent: :destroy
  has_many :power_supplies
  # has_many :cooler_fan_slots
  # has_many :case_fan_slots
  has_many :product_conflicts, as: :product, dependent: :destroy
  has_many :sources, as: :product, dependent: :destroy
end
